### DISCUSSION TITLE
how to find a supported texture format

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3513] [Q&A] [answered]
Is there a way to find a supported texture format other than just trying everything?  I'm looking for an RGB format for Mac/Metal, and so far I've found that RGB8UI, RGB8I, RGB32UI, RGB32I, and RGB32F are unsupported.

**Comment by bejado** [comment, marked as answer]
For Metal at least, you can check here: https://github.com/google/filament/blob/00f50c5f5a69feed46aa038544e420f2c458aba2/filament/backend/src/metal/MetalEnums.h#L129
Metal doesn't support 3-component textures natively.

**Reply by jwwalker** [reply]
I was assuming that it would be asking for trouble to use a format with alpha if I'm not actually going to use alpha.  Is it safe to use an RGBA format and just let the alpha channel take up space without doing anything?

**Reply by romainguy** [reply]
Yes. And because of alignment reasons RGB formats are often expanded to RGBA internally anyway.
